tests/Microsoft.Identity.Test.Unit.net45/PublicApiTests/AcquireTokenSilentTests.cs

```diff
@@ -46,6 +47,59 @@ public async Task NullAccount_EmptyLoginHintAsync()
 
         }
 
+        [TestMethod]
+        public async Task BadTokenError_RemoveAccountFromCacheAsync()
+        {
+            // Arrange
+            using (MockHttpAndServiceBundle harness = base.CreateTestHarness())
+            {
+                Trace.WriteLine("1. Create PCA");
+
+                PublicClientApplication app = PublicClientApplicationBuilder.Create(TestConstants.ClientId)
+                                                                            .WithHttpManager(harness.HttpManager)
+                                                                            .BuildConcrete();
+
+                var tokenCacheHelper = new TokenCacheHelper();
+                tokenCacheHelper.PopulateCache(app.UserTokenCacheInternal.Accessor, addSecondAt: false);
+
+                Trace.WriteLine("2. Configure AAD to respond with the typical Invalid Grant error and Bad Token sub error");
+                AddHttpMocks(harness.HttpManager);
+                var account = new Account(TestConstants.s_userIdentifier, TestConstants.DisplayableId, null);
+
+                // Act
+                var excepion = await AssertException.TaskThrowsAsync<MsalUiRequiredException>(() => app
+                    .AcquireTokenSilent(
+                        TestConstants.s_scope.ToArray(),
+                        account)
+                    .WithForceRefresh(true)
+                    .ExecuteAsync())
+                    .ConfigureAwait(false);
+
+                Assert.AreEqual(MsalError.BadToken, excepion.SubError);
+
+                var accounts = await app.GetAccountsAsync().ConfigureAwait(false);
+                Assert.IsFalse(accounts.Any());
+
+                var refreshTokens = app.UserTokenCacheInternal.Accessor.GetAllRefreshTokens();
+                Assert.IsFalse(refreshTokens.Any());
+            }
+        }
+
+        private static void AddHttpMocks(MockHttpManager httpManager)
```

Comment:
>AddHttpMocks [](start = 28, length = 12)

since this mock actually creates a invalid_grant error, could you modify the name to include that? it's a little misleading if just reading the code from the test that the invalid_grant part will be configured here.

Reply:
Good point. Updated the name.